diff -urN elliptics-2.26.11.0.orig/bindings/python/dnet_balancer elliptics-2.26.11.0/bindings/python/dnet_balancer
--- elliptics-2.26.11.0.orig/bindings/python/dnet_balancer	2017-01-12 14:43:56.194378821 +0300
+++ elliptics-2.26.11.0/bindings/python/dnet_balancer	2017-01-12 14:44:30.716947637 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # =============================================================================
 # 2013+ Copyright (c) Kirill Smorodinnikov <shaitkir@gmail.com>
diff -urN elliptics-2.26.11.0.orig/recovery/dnet_recovery elliptics-2.26.11.0/recovery/dnet_recovery
--- elliptics-2.26.11.0.orig/recovery/dnet_recovery	2017-01-12 14:43:56.188378722 +0300
+++ elliptics-2.26.11.0/recovery/dnet_recovery	2017-01-12 14:44:07.580566427 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # =============================================================================
 # 2013+ Copyright (c) Alexey Ivanov <rbtz@ph34r.me>
